File: ios/liveactivitytimer/TimerEventEmitter.m
//
//  TimerEventEmitter.m
//  liveactivitytimer
//
//  Created by Vitor Scheffer on 19/11/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(TimerEventEmitter, RCTEventEmitter)

+ (bool)requiresMainQueueSetup {
  return NO;
}

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/TimerEventEmitter.swift
//
//  TimerEventEmitter.swift
//  liveactivitytimer
//
//  Created by Vitor Scheffer on 19/11/24.
//

import Foundation

@objc(TimerEventEmitter)
class TimerEventEmitter: RCTEventEmitter {

  public static var emitter: TimerEventEmitter?

  override init() {
    super.init()
    TimerEventEmitter.emitter = self
  }

  override func supportedEvents() -> [String]! {
    return ["onPause", "onResume", "onRestart", "onReset", "onFinish"]
  }
}
